packages/tooltip/src/vaadin-tooltip.js

```diff
@@ -295,143 +89,10 @@ class Tooltip extends OverlayClassMixin(ThemePropertyMixin(ElementMixin(Controll
-       * or detecting whether to show the tooltip or not.
-       */
-      context: {
-        type: Object,
-        value: () => {
-          return {};
-        },
-      },
-
-      /**
-       * The delay in milliseconds before the tooltip
-       * is opened on keyboard focus, when not in manual mode.
-       * @attr {number} focus-delay
-       */
-      focusDelay: {
-        type: Number,
-      },
-
-      /**
-       * The id of the element used as a tooltip trigger.
-       * The element should be in the DOM by the time when
-       * the attribute is set, otherwise a warning is shown.
-       */
-      for: {
-        type: String,
-        observer: '__forChanged',
-      },
-
-      /**
-       * The delay in milliseconds before the tooltip
-       * is closed on losing hover, when not in manual mode.
-       * On blur, the tooltip is closed immediately.
-       * @attr {number} hide-delay
-       */
-      hideDelay: {
-        type: Number,
-      },
-
-      /**
-       * The delay in milliseconds before the tooltip
-       * is opened on hover, when not in manual mode.
-       * @attr {number} hover-delay
-       */
-      hoverDelay: {
-        type: Number,
-      },
-
-      /**
-       * When true, the tooltip is controlled programmatically
-       * instead of reacting to focus and mouse events.
-       */
-      manual: {
-        type: Boolean,
-        value: false,
-      },
-
-      /**
-       * When true, the tooltip is opened programmatically.
-       * Only works if `manual` is set to `true`.
-       */
-      opened: {
-        type: Boolean,
-        value: false,
-      },
-
-      /**
-       * Position of the tooltip with respect to its target.
-       * Supported values: `top-start`, `top`, `top-end`,
-       * `bottom-start`, `bottom`, `bottom-end`, `start-top`,
-       * `start`, `start-bottom`, `end-top`, `end`, `end-bottom`.
-       */
-      position: {
-        type: String,
-      },
-
-      /**
-       * Function used to detect whether to show the tooltip based on a condition,
-       * called every time the tooltip is about to be shown on hover and focus.
-       * The function takes two parameters: `target` and `context`, which contain
-       * values of the corresponding tooltip properties at the time of calling.
-       * The tooltip is only shown when the function invocation returns `true`.
-       */
-      shouldShow: {
-        type: Object,
-        value: () => {
-          return (_target, _context) => true;
-        },
-      },
-
-      /**
-       * Reference to the element used as a tooltip trigger.
-       * The target must be placed in the same shadow scope.
-       * Defaults to an element referenced with `for`.
-       */
-      target: {
-        type: Object,
-        observer: '__targetChanged',
-      },
-
-      /**
-       * String used as a tooltip content.
-       */
-      text: {
+      /** @private */
+      __effectivePosition: {
```

Comment:
Why not also move the members that are used in the template? I'd assume they can be used the same with a Lit template.

Reply:
Done.